regress/pesterTests/FileBasedLogging.tests.ps1

```diff
@@ -0,0 +1,237 @@
+        $newContent | Set-Content -Path $sshdconfig_custom
+				   
+        #skip when the task schedular (*-ScheduledTask) cmdlets does not exist
+        $ts = (get-command get-ScheduledTask -ErrorAction SilentlyContinue)
+        $skip = $ts -eq $null
+        if(-not $skip)
+        {
+            Stop-SSHDTestDaemon   -Port $port
+        }
+        if(($platform -eq [PlatformType]::Windows) -and ([Environment]::OSVersion.Version.Major -le 6))
+        {
+            #suppress the firewall blocking dialogue on win7
+            netsh advfirewall firewall add rule name="sshd" program="$($OpenSSHTestInfo['OpenSSHBinPath'])\sshd.exe" protocol=any action=allow dir=in
+        }
+    }
+
+    AfterEach { $tI++ }
+    
+    AfterAll {        
+        if(($platform -eq [PlatformType]::Windows) -and ($psversiontable.BuildVersion.Major -le 6))
+        {            
+            netsh advfirewall firewall delete rule name="sshd" program="$($OpenSSHTestInfo['OpenSSHBinPath'])\sshd.exe" protocol=any dir=in
+        }    
+    }
+
+
+    Context "Tests Logs for SSH connections" {
+        BeforeAll {            
+            $sshdConfigPath = $sshdconfig_custom
+
+            Add-PasswordSetting -Pass $password
+
+            $tI=1
+        }
+        
+        BeforeEach {
+            $sshlog = Join-Path $testDir "$tC.$tI.$sshLogName"            
+            $sshdlog = Join-Path $testDir "$tC.$tI.$sshdLogName"
+
+            if (Test-Path $sshdlog -PathType Leaf) {
+                Clear-Content $sshdlog
+            }
+
+            if(-not $skip)
+            {
+                Stop-SSHDTestDaemon   -Port $port
+            }
+        }
+
+        AfterAll {            
+            Remove-PasswordSetting
+            $tC++
+        }
+
+        It "$tC.$tI-Nonadmin SSH Connection"  -skip:$skip {
+            Start-SSHDTestDaemon -WorkDir $opensshbinpath -Arguments "-ddd -f $sshdConfigPath -E $sshdlog" -Port $port
+            $o = ssh -vvv -p $port -E $sshlog $nonadminusername@$server echo 1234
+            $o | Should Be 1234
+            Stop-SSHDTestDaemon   -Port $port
+            $sshdlog | Should Contain "KEX done \[preauth\]"
+            $sshdlog | Should Contain "exec_command: echo 1234"
+        }
+
+        It "$tC.$tI-Admin SSH Connection"  -skip:$skip {
+            Start-SSHDTestDaemon -WorkDir $opensshbinpath -Arguments "-ddd -f $sshdConfigPath -E $sshdlog" -Port $port
+            $o = ssh -vvv -p $port -E $sshlog $adminusername@$server echo 1234
+            $o | Should Be 1234
+            Stop-SSHDTestDaemon   -Port $port
+            $sshdlog | Should Contain "KEX done \[preauth\]"
+            $sshdlog | Should Contain "exec_command: echo 1234"
+        }
+
+    }
+	
+    Context "Tests Logs for SFTP connections" {
+		
+        BeforeAll {
+
+            $sshdConfigPath = $sshdconfig_custom
+			
+            function Setup-KeyBasedAuth
```

Comment:
What's the need for doing this? Why can't it be same as sshd?

Reply:
I wanted to use that -b flag to include a batch file with a couple of commands in the sftp tests, but that only works if we use it with non-interactive authentication, so that's why I has to use key based authentication for the sftp tests.